src/mds/MDSMap.cc

```diff
@@ -487,6 +487,7 @@ void MDSMap::encode(bufferlist& bl, uint64_t features) const
     ::encode(session_autoclose, bl);
     ::encode(max_file_size, bl);
     ::encode(max_mds, bl);
+    ::encode(balancer, bl);
```

Comment:
This is the legacy encoding for clients too old to handle versioned things, don't add balancer here